bsip-0084.md

```diff
@@ -0,0 +1,195 @@
+
+The flag must not be used before the time of the hardfork.
+
+
+## 2. New operation "create_elected_authority"
+
+**Note 1:** Since the overall computational overhead for this voting mechanism is significant, the height of the fee should reflect this.
+
+**Note 2:** The new asset flag `voting_allowed` is only checked for this operation. If the flag is removed from an asset, any existing elected authorities are unaffected.
+
+**Fields:**
+
+* `account_id_type creator` - the account to pay the fee, also the only one who can delete the authority
+* `asset_id_type voting_asset` - the asset on which to base the voting
+* `unsigned_int num_members` - the number of authority members to vote on if fixed, or 0 otherwise
+* `unsigned_int min_members` - the minimum number of authority members to elect, or 0 if fixed
+* `unsigned_int max_members` - the maximum number of authority members that can be elected, or 0 if fixed
+* `flat_set<account_id_type> candidates` - a list of candidates eligible for voting, or empty
+* `optional<asset_id_type> candidate_holders` - an asset that candidates must hold to be eligible for voting
+* `bool proxy_allowed` - indicates if proxy voting is allowed
+* `optional<time_point_sec> vote_until` - an optional ending date after which voting slates are frozen
+
+**Validation:**
+
+The operation must not be used before the time of the hardfork.
+
+* `creator` must exist, must have lifetime membership, and must have sufficient balance to pay the fee.
+* `voting_asset` must exist and must have the `voting_allowed` flag set.
+* If `num_members` is 0 then `min_members` and `max_members` must both be positive.
+* If `num_members` is positive then both `min_members` and `max_members` must equal 0.
+* If `candidates` is not empty then
+    * Its size must be greater or equal to `max(num_members,min_members)`.
+    * `candidate_holders` must not be present.
+* If `candidate_holders` is present the `candidates` must be empty.
+* `vote_until`, if present, must be in the future
+
+**Evaluation:**
+
+A new object type `elected_authority_object` is introduced. The operation creates such an object using the fields from the operation.
+
+A new, empty authority is created. The desired number of members for the authority is set to `max(num_members,min_members)`.
+
+Half of the operation fee is set aside and stored in the `elected_authority_object`.
+
+
+## 3. New operation "delete_elected_authority"
+
+**Fields:**
+
+* `account_id_type owner` - the account to pay the fee, must have created the authority
+* `elected_authority_id_type authority` - the authority to delete
+
+**Validation:**
+
+The operation must not be used before the time of the hardfork.
+
+* `authority` must exist.
+* `owner == authority.creator`
+
+**Evaluation:**
+
+* The fee stored in `authority` is returned to `owner`.
+* The `elected_authority_object` is deleted.
+* All related objects, including the `authority` as well as all user votes, are also deleted.
```

Comment:
I think this gives the creator too big power in some scenarios, it seems too risky for the accounts that has the elected authority as owner/active, as well as other entities that utilize the elected authority.

Reply:
Yes, I have thought about how to handle this. The problem is, if we disallow deletion if the authority is in use anywhere, then it is easy for someone to block an authority from being deleted. 

OTOH I think someone who uses an elected authority that is not under his control can be expected to coordinate this with the authority owner. Setting an elected authority as owner on an account is a bad idea, because as you say, authority deletion would block the account. Use as an active authority (or feed producer) can easily be repaired by the account owner (or asset owner).

I therefore think it is better to delete the authority anyway, and let those who are using it deal with it.

Reply:
Please add these to the discussions section and/or a risks section. Thanks.

Reply:
Done.